File: src/OddEvenSort.java
public class OddEvenSort {
    private boolean isOdd(int value) {
        return (value & 1) == 1;
    }

    public void oddEvenSort(int[] a, int n) {
        int i = 0;
        int j = n - 1;
        while (i < j) {
            while (isOdd(a[i])) {
                i++;
            }
            while (!isOdd(a[j])) {
                j--;
            }
            if (i < j) {
                int temp = a[i];
                a[i] = a[j];
                a[j] = temp;
                i++;
                j--;
            }
        }
    }

    public static void main(String[] args) {
        int[] a = { 2, 2, 3, 4, 5, 7 };
        OddEvenSort oSort = new OddEvenSort();
        oSort.oddEvenSort(a, 6);
        for (int value : a) {
            System.out.print(value);
            System.out.print(" ");
        }
    }
}

File: src/javascript/odd-even-sort.js
function isOdd(value) {
  return (value & 1) == 1;
}

function oddEvenSort(arr) {
  let i = 0;
  let j = arr.length - 1;
  while (i <= j) {
    while (isOdd(arr[i])) {
      i++;
    }
    while (!isOdd(arr[j])) {
      j--;
    }
    if (i <= j) {
      let temp = arr[i];
      arr[i] = arr[j];
      arr[j] = temp;
      i++;
      j--;
    }
  }
}

var arr = [
  2,
  2,
  3,
  4,
  5,
  7
];
oddEvenSort(arr);
console.log(arr);